### ISSUE TITLE
feature request: variables and parameters in different colours

### ISSUE THREAD

**Issue by pierfle** [MiniZinc/MiniZincIDE#115]
I suspect this is difficult to do, but it would be nice if the IDE had (one day) an option to display all the decision variables in one colour and all the parameters in another colour: hunting for common symptoms of inefficiency or potentially unexpected behaviour (such as unfixed `where` clauses) in a model would be easier.

**Comment by guidotack** [comment]
I agree this would be nice. However it requires full incremental type checking of the model, and the type checking would have to be reasonably robust against syntax and type errors (the current parser and type checker just give up when anything is incorrect). I guess it would be possible to make this work for models that have no syntax or type errors, and fall back on the current syntax highlighting as soon as an error is detected.

**Comment by pierfle** [comment]
What about non-incremental colouring: just on-demand, with frozen editing?
